### PR TITLE
add badges on how many objects we have in a list

### PR DESCRIPTION
**Pull request by jens1o**
Provides some kind of consistency. 

When this gets merged, you're free to do that too with your applications. :)

For those who wonder where that `$icons` variable is given to the template, see `MultipleLinkPage` (which is inherited from SortablePage, which is inherited from AbstractListPage)

### CODE REVIEW

**Comment by jacboy** [comment]
Up for that!

File: com.woltlab.wcf/templates/trophyList.tpl [hunk 1]
```diff
@@ -3,7 +3,7 @@
 {capture assign='contentHeader'}
 	<header class="contentHeader messageGroupContentHeader">
 		<div class="contentHeaderTitle">
-			<h1 class="contentTitle">{$category->getTitle()}</h1>
+			<h1 class="contentTitle">{$category->getTitle()} <span class="badge badgeInverse">{#$items}</span></h1>
```
Comment (dtdesign) [inline com.woltlab.wcf/templates/trophyList.tpl]: The badge is shown regardless if there are any items or not, please wrap the span inside `{if $items}…{/if}`. This applies to  other templates too, but is not required for those that will always contain items (such as users, groups or packages).

File: wcfsetup/install/files/acp/templates/articleList.tpl [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 <header class="contentHeader">
 	<div class="contentHeaderTitle">
-		<h1 class="contentTitle">{lang}wcf.acp.article.list{/lang}</h1>
+		<h1 class="contentTitle">{lang}wcf.acp.article.list{/lang}  <span class="badge badgeInverse">{#$items}</span></h1>
```
Comment (dtdesign) [inline wcfsetup/install/files/acp/templates/articleList.tpl]: Superfluous whiespace between the title and the badge. Applies to some other changes too.

**Comment by jens1o** [comment]
Okay, I will do the requested changes, but atm, I'm not at my coding machine.

File: com.woltlab.wcf/templates/trophyList.tpl (hunk 1 above)
Comment (jens1o) [inline com.woltlab.wcf/templates/trophyList.tpl]: Okay, I thought it's fine to show that they are zero items.

Comment (dtdesign) [inline com.woltlab.wcf/templates/trophyList.tpl]: Neither do we always show it for zero values, nor do we always hide it. It depends on the context, in particular if the `0` has an actual meaning.

For example, the draft section in conversations do display a zero to indicate that there are no drafts, avoiding confusing with numbers only seen when there are unread items. But in case of the trophy list (and others) there is already a clear indication, that there are absolutely no items, therefore displaying the zero has no purpose. Also you're already on that very page, it is obvious that there are no items.

**Comment by dtdesign** [comment]
@jens1o One week has passed, are you still working on it?

**Comment by jens1o** [comment]
Well, time has passed and I got my machine back today... will finish it :)

**Comment by jens1o** [comment]
just committed my changes, hope they fit your needs.

File: com.woltlab.wcf/templates/trophyList.tpl (hunk 1 above)
Comment (jens1o) [inline com.woltlab.wcf/templates/trophyList.tpl]: Somehow GitHub didn't recognized I had changed that file? In my local file this has been fixed?

Comment (dtdesign) [inline com.woltlab.wcf/templates/trophyList.tpl]: I somehow doubt that, maybe run a push again to see if there is something going on.

File: wcfsetup/install/files/acp/templates/languageItemList.tpl [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 <header class="contentHeader">
 	<div class="contentHeaderTitle">
-		<h1 class="contentTitle">{lang}wcf.acp.language.item.list{/lang}</h1>
+		<h1 class="contentTitle">{lang}wcf.acp.language.item.list{/lang} <span class="badge badgeInverse">{#$items}</span></h1>
```
Comment (dtdesign) [inline wcfsetup/install/files/acp/templates/languageItemList.tpl]: This can actually be an empty page if no items match the search criteria.

File: wcfsetup/install/files/acp/templates/userOptionList.tpl [hunk 4]
```diff
@@ -9,7 +9,7 @@
 
 <header class="contentHeader">
 	<div class="contentHeaderTitle">
-		<h1 class="contentTitle">{lang}wcf.acp.user.option.list{/lang}</h1>
+		<h1 class="contentTitle">{lang}wcf.acp.user.option.list{/lang}{if $items} <span class="badge badgeInverse">{#$items}</span>{/if}</h1>
```
Comment (dtdesign) [inline wcfsetup/install/files/acp/templates/userOptionList.tpl]: This list can never be empty.

File: com.woltlab.wcf/templates/trophyList.tpl (hunk 1 above)
Comment (dtdesign) [inline com.woltlab.wcf/templates/trophyList.tpl]: $parameters = $conditions->getParameters();

Comment (jens1o) [inline com.woltlab.wcf/templates/trophyList.tpl]: I know, and I fixed it here... I'm really sure... And I can't fix that somehow, because at my local version, this already has been fixed and git won't commit that?

Comment (dtdesign) [inline com.woltlab.wcf/templates/trophyList.tpl]: Strange, it looks okay when reviewing the file in your repository: https://github.com/jens1o/WCF/commit/3f08bfd8b26122d3f322808eb5a39132263d3531#diff-0f077676cc079dc0e9ea7493ec05a6d3

Comment (jens1o) [inline com.woltlab.wcf/templates/trophyList.tpl]: Oh... I missed that this template is not in the acp...

Comment (dtdesign) [inline com.woltlab.wcf/templates/trophyList.tpl]: Oh right, that one slipped through. In that case, please revert the change to this particular template.

**Comment by jens1o** [comment]
like this?

File: wcfsetup/install/files/acp/templates/userList.tpl [hunk 5]
```diff
@@ -56,7 +56,7 @@
 
 <header class="contentHeader">
 	<div class="contentHeaderTitle">
-		<h1 class="contentTitle">{lang}{@$pageTitle}{/lang} <span class="badge badgeInverse">{#$items}</span></h1>
```
Comment (dtdesign) [inline wcfsetup/install/files/acp/templates/userList.tpl]: Why did you remove the counter _and_ the `<h1>`? Please revert this change.

Comment (jens1o) [inline wcfsetup/install/files/acp/templates/userList.tpl]: That wasn't me!111

File: wcfsetup/install/files/acp/templates/userList.tpl [hunk 6]
```diff
@@ -56,7 +56,7 @@
 
 <header class="contentHeader">
 	<div class="contentHeaderTitle">
-		<h1 class="contentTitle">{lang}{@$pageTitle}{/lang}
+		<h1 class="contentTitle">{lang}{@$pageTitle}{/lang}</h1>
```
Comment (dtdesign) [inline wcfsetup/install/files/acp/templates/userList.tpl]: Well, you fixed the markup, but the counter is still missing. Also, because it supports filtering, this list can be empty, requiring the counter to be hidden at times.

Comment (jens1o) [inline wcfsetup/install/files/acp/templates/userList.tpl]: I thought I shall remove it because there always are some users. I shouldn't be that hectic. Sorry.
